Whoaverse/Whoaverse/Controllers/HomeController.cs

```diff
@@ -398,6 +398,59 @@ public ActionResult UserProfile(string id, int? page, string whattodisplay)
                 return View("UserSubmitted", paginatedUserSubmissions);
             }
 
+            // show saved                        
+            if (whattodisplay != null && whattodisplay == "saved" && User.Identity.IsAuthenticated && User.Identity.Name == id)
+            {
+                var savedSubmissions = _db.Savingtrackers.Where(u => u.UserName == id).ToDictionary(saveSubTracker => saveSubTracker.MessageId, saveSubTracker => saveSubTracker.Timestamp);
+                var savedComments = _db.Commentsavingtrackers.Where(u => u.UserName == id).ToDictionary(saveCommentTracker => saveCommentTracker.CommentId, saveCommentTracker => saveCommentTracker.Timestamp);
+
+                var submissionIds = new List<int>(savedSubmissions.Keys);
+                var commentIds = new List<int>(savedComments.Keys);
+
+                var submissionsSavedByUser = _db.Messages.Where(submission => submissionIds.Contains(submission.Id)).ToList();
+
+                var commentsSavedByUser = _db.Comments.Where(comment => commentIds.Contains(comment.Id)).ToList();
+
+                //Merge submissions and comments into one list sorted by date
```

Comment:
I don't think that will work as the timestamp is stored on the saving tracker object, not the Comment or Message object. I'm not sorting by the time the comment is created, which is what the comment/submission has stored on it, but instead by the time the user saved it. Unless I'm misunderstanding that SO link, that is all I would be able to do with the merged list is access the properties of a Comment/Message object. Since there has to be a lookup of submissions/comments found in the SavingTrackers (which stores a user, message id, and saving timestamp), I don't think this can be done in a LINQ to Entities query.

Step by step of what my code does:
1. Fetch submission/comment saving trackers associated with the logged in user from the database.
2. Create a dictionary mapping each saving tracker's message id to its timestamp (for quick lookup later)
3. Find all submissions/comments with the found message ids
4. Using the id to timestamp Dictionary, merge and sort the submission list and comment list by the time that they were saved, not created.

Once again, I'm new to C#, so if this can be done with LINQ, could you point out how?